### PR TITLE
Run app with nodemon

### PR DESCRIPTION
**Pull request by msm-faraj**
This pr aims to:
install nodemon globlay in the server container with change of dockerfile.
and change the "package.json" file to run the app with nodemon.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,7 @@
 FROM node:21-alpine
 
+RUN npm install -g nodemon
```
Comment (pouyapourbaba) [inline Dockerfile:3]: Install this as a dev dependency and have it listed in the package.json  https://docs.npmjs.com/specifying-dependencies-and-devdependencies-in-a-package-json-file#adding-dependencies-to-a-packagejson-file-from-the-command-line

`npm i nodemon --save-dev`

File: package.json [hunk 2]
```diff
@@ -4,18 +4,18 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "dev": "cd db && sequelize-cli db:migrate && cd .. && node src/index.js ",
+    "dev": "sequelize-cli db:migrate  && nodemon src/index.js ",
```
Comment (pouyapourbaba) [inline package.json:7]: For now this is fine, but, create an issue on the Kanban board to move the `sequelize-cli db:migrate` part inside the application.

File: Dockerfile (hunk 1 above)
Comment (msm-faraj) [reply Dockerfile:3]: done

Comment (msm-faraj) [reply Dockerfile:3]: This was really good because now this extra line removed and nodemon installed with RUN npm in dockerfile. Great

Comment (pouyapourbaba) [reply Dockerfile:3]: Where is the change then? I do not see nodemon in package.josn. 
Plus, you do not need this line here anymore.

File: package.json [hunk 3]
```diff
@@ -4,18 +4,18 @@
...
-    "pg-hstore": "^2.3.4",
-    "sequelize": "^6.37.3"
+    "express": "4.19.2",
+    "pg": "8.12.0",
+    "pg-hstore": "2.3.4",
+    "sequelize": "6.37.3"
   },
   "devDependencies": {
-    "sequelize-cli": "^6.6.2"
+    "sequelize-cli": "6.6.2"
```
[the remark this answers is not in the document]
Comment (pouyapourbaba) [reply package.json:19]: On the second thought, let it be here, no need to remove it.

File: Dockerfile (hunk 1 above)
Comment (msm-faraj) [reply Dockerfile]: now you can see the changes

File: package.json [hunk 4]
```diff
@@ -4,18 +4,19 @@
...
-    "pg-hstore": "^2.3.4",
-    "sequelize": "^6.37.3"
+    "express": "4.19.2",
+    "pg": "8.12.0",
+    "pg-hstore": "2.3.4",
+    "sequelize": "6.37.3"
   },
   "devDependencies": {
-    "sequelize-cli": "^6.6.2"
+    "nodemon": "^3.1.4",
```
Comment (pouyapourbaba) [inline package.json:19]: Pin this dependency by removing the `^` character and running `npm install`

**Comment by pouyapourbaba** [comment]
One more thing, let's rebase this branch on the latest master. The changes from the other PR are in this PR now.

**Review by pouyapourbaba — APPROVED** [review]

**Pull request merged**